### PR TITLE
Fixes #12 - Improved Api#fetchChangesSince error messages.

### LINKED ISSUE #12: Improve error reporting when fetching changes failed.
```diff
   71            return {items: []};
   72          } else if (res.status >= 400) {
   73:           // TODO: attach better error reporting
   74            throw new Error("Fetching changes failed: HTTP " + res.status);
   75          } else {
```
We'd need to distinguish between possible 4xx and 5xx status codes so the end user knows more precisely what happened.

### PR DESCRIPTION
**Pull request by n1k0**
Refs #12.

r=? @Natim @leplatrem @ametaireau

### CODE REVIEW

File: src/api.js [hunk 1]
```diff
@@ -79,21 +89,27 @@ export default class Api {
 
     return fetch(recordsUrl + queryString, {headers})
       .then(res => {
+        response = res;
```
Comment (Natim) [inline src/api.js]: can't we use response directly instead of res?

Comment (n1k0) [inline src/api.js]: You mean using res directly? We're assigning the result to the response var available from the outter scope, so next Promise steps can still access the response object.

Comment (Natim) [inline src/api.js]: Oh ok yes you are right.

**Comment by n1k0** [comment]
Let me land that baby.

**Pull request merged**